File: archive/deploy-0315/terraform-lambda/market_prices/function.js
const aws = require("aws-sdk");

const s3 = new aws.S3({ apiVersion: "2006-03-01" });

// Function to convert JSON to CSV

function randomIntFromInterval(min, max) {
  // min and max included
  return Math.random() * (max - min) + min;
}

exports.handler = async (event) => {
  console.log("Event: ", event);
  let responseMessage = "market-prices!";

  if (event.httpMethod !== "POST") {
    return {
      statusCode: 401,
      body: JSON.stringify({ message: "Invalid or missing token" }),
    };
  }

  const market_prices = randomIntFromInterval(0.1, 0.25).toFixed(2);
  // console.log(rndInt);
  const data = { market_prices: market_prices };
  const response = {
    statusCode: 200,
    headers: {
      "Content-Type": "application/json",
    },
    body: JSON.stringify({
      message: JSON.stringify(data),
    }),
  };

  return response;
};

File: terraform-lambda/delete_me/markets/function.py
import json


def handler(event, context):
    """
    Get /all-markets-status, Get the status of all markets
    get_all_markets_status_handler
    """

    """
    Get /db/markets, Get a list of markets.
    list_markets_handler
    """
    """
    GET /db/market/<market_id>, Get a market by market_id
    get_market_handler
    """
    """
    put_market_handler
    """
    """
    delete_market_handler
    """
    return {
        'statusCode': 200,
        'headers': {'Content-Type': 'application/json'},
        'body': json.dumps({'message': 'Hello from market lambda!'})
    }
